pkg/build/strategies/sti/sti.go

```diff
@@ -218,6 +219,17 @@ func (b *STI) PostExecute(containerID, location string) error {
 
 	buildEnv := append(scripts.ConvertEnvironment(env), b.generateConfigEnv()...)
 
+	labels := map[string]string{}
+	if b.sourceInfo != nil {
+		labels = map[string]string{
+			"io.openshift.build.commit.author":     b.sourceInfo.Author,
+			"io.openshift.build.commit.date":       b.sourceInfo.Date,
+			"io.openshift.build.commit.id":         b.sourceInfo.ID,
+			"io.openshift.build.commit.ref":        b.sourceInfo.Ref,
+			"io.openshift.build.source-repository": b.sourceInfo.Location,
```

Comment:
I'm a little bit concerned that source-repository here isn't the right naming, since you can build from local directory, which is not a source-repository (btw. how should we handle relative vs absolute paths then? always expand to absolute path?) One other issue is context directory, that is missing in here and it **has** to be here imho.